### DISCUSSION TITLE
No Hbond at all

### DISCUSSION THREAD

**Discussion by icamps** [MDAnalysis/mdanalysis#4410] [General]
Hello,

I ran a hydrogen bond analysis in VMD and PLIP. Both of them reported several bonds through the whole trajectory.

Now, I am running the hydrogen bond analysis with MDAnalysis which is reporting zero bond through the whole trajectory.

Here is the info about my software:
```
> Library version

 * Numpy      : 1.24.1
 * MDAnalysis : 2.6.1
 * Parmed     : 4.2.2
 * nglview    : 3.0.3
```

And the piece of code I am using:

```
from MDAnalysis.analysis.hydrogenbonds.hbond_analysis import HydrogenBondAnalysis as HBA

sel_prot = "segid PROA"
sel_lig = "segid HETA"
hbonds = HBA(
    universe=u_aligned,
    donors_sel=None,
    hydrogens_sel=None,
    acceptors_sel=None,
    d_a_cutoff=4.0,
    d_h_a_angle_cutoff=150,
    between=[sel_prot,sel_lig]
)
hbonds.run(verbose=True)

column_names = ["Frame", "Donor_id", "Hydrogen_id", "Acceptor_id", "Distance_Angs", "Angle_degree"]
hb = pd.DataFrame(hbonds.hbonds, columns = column_names)
print(hb)
tau_timeseries, timeseries = hbonds.lifetime()

hb.to_csv("out_" + fname_root + "_HBonds_Analysis" + ".csv")
```

The output is:

```
HBond Calculation...
100% 3000/3000 [01:04<00:00, 47.54it/s]

Empty DataFrame
Columns: [Frame, Donor_id, Hydrogen_id, Acceptor_id, Distance_Angs, Angle_degree]
Index: []
Done.
```

Any help is welcome.

**Comment by orbeckst** [comment]
What is HETA? If it contains non-standard atoms then guessing of donors/acceptors might not work.

Do `guess_hydrogens()` and `guess_acceptors()` (see https://docs.mdanalysis.org/stable/documentation_pages/analysis/hydrogenbonds.html#example-use-of-hydrogenbondanalysis) give the correct atoms?

Have you tried it with protein/water or protein/protein?

**Reply by icamps** [reply]
HETA is the segment name of the ligand, and PROA is the corresponding for the protein.

`guess_hydrogens()` returns empty, and `guess_acceptors()` returns:

`'(resname ALA and name O) or (resname ALA and name OT1) or (resname ALA and name OT2) or (resname ARG and name NE) or (resname ARG and name NH1) or (resname ARG and name NH2) or (resname ARG and name O) or (resname ASN and name ND2) or (resname ASN and name O) or (resname ASN and name OD1) or (resname ASP and name O) or (resname ASP and name OD1) or (resname ASP and name OD2) or (resname CYS and name O) or (resname GLN and name NE2) or (resname GLN and name O) or (resname GLN and name OE1) or (resname GLU and name O) or (resname GLU and name OE1) or (resname GLU and name OE2) or (resname GLY and name O) or (resname HSD and name NE2) or (resname HSD and name O) or (resname ILE and name O) or (resname LEU and name O) or (resname LIG and name N12) or (resname LIG and name N6) or (resname LIG and name O19) or (resname LYS and name O) or (resname MET and name O) or (resname PHE and name O) or (resname PRO and name O) or (resname SER and name O) or (resname SER and name OG) or (resname THR and name O) or (resname THR and name OG1) or (resname TRP and name NE1) or (resname TRP and name O) or (resname TYR and name O) or (resname TYR and name OH) or (resname VAL and name O)'`

Also, I ran the commands below:

```
# input
hbonds.hydrogens_sel = hbonds.guess_hydrogens("segid PROA")
hbonds.acceptors_sel = hbonds.guess_acceptors("segid PROA")
print(f"hydrogen_sel = {hbonds.hydrogens_sel}")
print(f"acceptors_sel = {hbonds.acceptors_sel}")
```
```
#output
hydrogen_sel = 
acceptors_sel = (resname ALA and name O) or (resname ALA and name OT1) or (resname ALA and name OT2) or (resname ARG and name NE) or (resname ARG and name NH1) or (resname ARG and name NH2) or (resname ARG and name O) or (resname ASN and name ND2) or (resname ASN and name O) or (resname ASN and name OD1) or (resname ASP and name O) or (resname ASP and name OD1) or (resname ASP and name OD2) or (resname CYS and name O) or (resname GLN and name NE2) or (resname GLN and name O) or (resname GLN and name OE1) or (resname GLU and name O) or (resname GLU and name OE1) or (resname GLU and name OE2) or (resname GLY and name O) or (resname HSD and name NE2) or (resname HSD and name O) or (resname ILE and name O) or (resname LEU and name O) or (resname LYS and name O) or (resname MET and name O) or (resname PHE and name O) or (resname PRO and name O) or (resname SER and name O) or (resname SER and name OG) or (resname THR and name O) or (resname THR and name OG1) or (resname TRP and name NE1) or (resname TRP and name O) or (resname TYR and name O) or (resname TYR and name OH) or (resname VAL and name O)

```
and

```
#input
hbonds.hydrogens_sel = hbonds.guess_hydrogens("segid HETA")
hbonds.acceptors_sel = hbonds.guess_acceptors("segid HETA")
print(f"hydrogen_sel = {hbonds.hydrogens_sel}")
print(f"acceptors_sel = {hbonds.acceptors_sel}")
```
```
#output
hydrogen_sel = 
acceptors_sel = (resname LIG and name N12) or (resname LIG and name N6) or (resname LIG and name O19)
```
The script I am using work fine with other systems (it is the same script) and the HBond analysis is based on the three links from [here](https://userguide.mdanalysis.org/stable/examples/analysis/README.html#hydrogen-bond-analysis).

Here are the links to the [PSF](https://drive.google.com/file/d/18L0h-TsZCTOKx4Rs_oxJ_Tqhv5ULpSVr/view?usp=drive_link) and [DCD](https://drive.google.com/file/d/1zCaJv4O-yR_Em_ypl-d15jZZy0hLQl6t/view?usp=sharing) files (in case you can take a look).

**Reply by orbeckst** [reply]
It looks as if your hydrogens are not automatically recognized. Have a look at how to specify the various selections. 

Sorry, I don't have the time to look at your actual files and figure it out.
